content/articles/linear and binary search in java/index.md

```diff
@@ -0,0 +1,269 @@
+images:
+  - url: /engineering-education/linear-and-binary-search-in-java-data-structures-and-algorithms/hero.jpg
+    alt: Data Structures cover image 
+---
+<!--more-->
+
+In the world of programming languages, data structures and algorithms are one of the problem-solving skills that all engineers must have. Linear and Binary Search are required when there is a problem with unsorted and sorted arrays in Java or any other language respectively. 
+
+In this article, I will be sharing the ways of utilizing the methods in solving interview questions. 
+
+#### Table of contents
+In the end, readers would know the following concepts;
+
+1. What are data structures and algorithms in Java.
+2. When and how to use linear search algorithm.
+3. Concepts of binary search in sorted java arrays problems.
+4. Solutions to interview questions
+
+### What are data structures and algorithms in Java.
+
+In Computer Science specifically, data structures have to do with the way data are been structured, modes of flow, storing of data, and methods of referencing. There are many ways of structuring data in programming languages and one of them is an array or list which is the point of discussion. 
+
+There are many methods of referencing the data stored through an array. Algorithms are the methods or stepwise rules laid down to solve the problem of data structures. 
+
+For instance, stack and queue are algorithms that describe the flow of data through an array or list in real-world applications. 
+
+Stack uses the algorithms of the last-in-first-out methodology which is called `LIFO`. This means that data are stacked on top of one another, while the last one will be the first to come up when calling. 
+
+However, Queue uses the analogy of first-in-first-out and is otherwise known as `FIFO`. Think of this as a public queue where the first person that came in will be the first one attended to. 
+
+There are others data structures and algorithms like liked-in lists, hash maps, and many more. However, we will be looking at binary and linear search algorithms for data stored in arrays.
+
+### Linear search algorithm
+
+Linear Search can be used to traverse through array problems. Like searching for a particular element, finding a maximum or minimum element e.t.c.
+
+Note that linear search algorithm is useful majorly when the array, in particular, is not sorted whether ascending or descending order. 
+
+Using a linear search algorithm requires looping through the entire array in `N` times. This means that the program will run through all elements of the array in unknown number of times. 
+
+More so, the time complexity of this algorithm shall be `O(N)`. The time complexity is said to be the time it takes the loop to run in exact length of the array.
+
+Fundamentally, the space complexity of the linear search algorithms in most cases is always `O(1)`. That is constant of one because the loop will run one time and no other memory space is consumed.
```

Comment:
```suggestion
Fundamentally, the space complexity of the linear search algorithms in most cases is always `O(1)`. That is, constant of one because the loop will run one time and no other memory space is consumed.
```